Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -150,19 +167,12 @@ void UPSWorldSubsystem::OnWorldBeginPlay(UWorld& InWorld)
 
 		if (GameFeatureName == "ProgressionSystem")
 		{
-			// Subscribe events on player type changed and Character spawned
-			BIND_ON_LOCAL_CHARACTER_READY(this, ThisClass::OnCharacterReady);
-
-			// Listen to handle input for each game state
-			BIND_ON_GAME_STATE_CHANGED(this, ThisClass::OnGameStateChanged);
-
-			LoadGameFromSave();
-
-			StarDynamicProgressMaterial = UMaterialInstanceDynamic::Create(UPSDataAsset::Get().GetDynamicProgressionMaterial(), this);
-			if (!ensureMsgf(StarDynamicProgressMaterial, TEXT("ASSERT: [%i] %hs:\n'StarDynamicProgressMaterial' is null!"), __LINE__, __FUNCTION__))
-			{
-				return;
-			}
+			OnInitialize.AddDynamic(this, &ThisClass::OnInitialized);
+			
+			// Load save game data of the Main Menu
+			FAsyncLoadGameFromSlotDelegate AsyncLoadGameFromSlotDelegate;
+			AsyncLoadGameFromSlotDelegate.BindUObject(this, &ThisClass::OnAsyncLoadGameFromSlotCompleted);
+			UGameplayStatics::AsyncLoadGameFromSlot(SaveGameDataInternal->GetSaveSlotName(), SaveGameDataInternal->GetSaveSlotIndex(), AsyncLoadGameFromSlotDelegate);
```

Comment:
Use static scope to call `GetSaveSlotName()` and `GetSaveSlotIndex()` directly, as these are static functions. This improves readability and reduces unnecessary pointer usage.

### Why
- **Improves readability**: Directly calls static functions, making it clear they don’t rely on an instance.
- **Removes redundancy**: Avoids the unnecessary use of a pointer to access static members.

### Expected result
```cpp
UGameplayStatics::AsyncLoadGameFromSlot(UPSSaveGameData::GetSaveSlotName(), UPSSaveGameData::GetSaveSlotIndex(), AsyncLoadGameFromSlotDelegate);

Reply:
done https://github.com/h4rdmol/ProgressionSystem/pull/21/commits/caee012b8a1c03aa2b807efa633a78563a096f83